numpyro/distributions/continuous.py

```diff
@@ -1680,6 +1680,70 @@ def infer_shapes(df, loc, scale_tril):
         return batch_shape, event_shape
 
 
+class GaussianCopula(Distribution):
+    arg_constraints = {
+        "covariance_matrix": constraints.positive_definite,
+        "precision_matrix": constraints.positive_definite,
+        "scale_tril": constraints.lower_cholesky,
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "covariance_matrix",
+        "precision_matrix",
+        "scale_tril",
+    ]
+
+    support = constraints.unit_interval
+
+    def __init__(
+        self,
+        transform_dist,
+        covariance_matrix=None,
+        precision_matrix=None,
+        scale_tril=None,
+        validate_args=None,
+    ):
+        self.transform_dist = transform_dist
+        self.base_dist = MultivariateNormal(
+            covariance_matrix=covariance_matrix,
+            precision_matrix=precision_matrix,
+            scale_tril=scale_tril,
+        )
+        self.normal = Normal(scale=jnp.diag(self.base_dist.covariance_matrix))
+
+        event_shape = self.base_dist.event_shape
+        delta_dim = len(event_shape) - len(transform_dist.event_shape)
+
+        batch_shape = lax.broadcast_shapes(
+            self.transform_dist.batch_shape[:-delta_dim],
+            self.base_dist.batch_shape,
+        )
+
+        super(GaussianCopula, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+
+        shape = sample_shape + self.batch_shape[: -len(self.base_dist.batch_shape)]
+        normal_samples = self.base_dist.sample(sample_shape=shape, key=key)
+        cdf = self.normal.cdf(normal_samples)
```

Comment:
Probably I misunderstand this - I guess we only need `cdf = dist.Normal(0, 1).cdf(normal_samples)` here?

Reply:
Oh, I see the reason. You are using covariance matrix, not correlation matrix. As mentioned in another comment, it's better to use correlation matrix to help inference over this parameter.